src/shell.js

```diff
@@ -177,6 +205,17 @@ if (ENVIRONMENT_IS_NODE) {
 #if ENVIRONMENT && ASSERTIONS
   if (typeof process == 'undefined' || !process.release || process.release.name !== 'node') throw new Error('not compiled for this environment (did you build to HTML and try to run it not on the web, or set ENVIRONMENT to something - like node - and run it someplace else - like on the web?)');
 #endif
+
+#if ASSERTIONS
+  var nodeVersion = process.versions.node;
+  var numericVersion = nodeVersion.split('.').slice(0, 3);
+  numericVersion = (numericVersion[0] * 10000) + (numericVersion[1] * 100) + numericVersion[2] * 1;
+  var minVersion = {{{ MIN_NODE_VERSION }}};
+  if (numericVersion < {{{ MIN_NODE_VERSION }}}) {
+    abort('This emscripten-generated code requires node {{{ formattedMinNodeVersion() }}} (detected v' + nodeVersion + ')');
```

Comment:
Is abort valid to call during startup? The code on line 206 uses `throw new Error` for a similar situation.

Reply:
I had to move some code around to make it valid, but yes.  That test verified that.  Perhaps I should update line 206 while I'm here?

Reply:
Updated all the `throw new Error` here to `abort`.